### PR TITLE
Add support for batching, error events and singleton instantiation

### PR DESCRIPTION
**Pull request by gunnarlium**
This allows clients to control when events will be sent to Grimm.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,11 +13,24 @@ Install with npm:
...
+const grimm = require('@aptoma/grimm-pusher');
+
+const options = {
+	host: 'http://grimm.example.com',
+	apiKey: 'secret',
+	// Use process decider to control when items are actually sent
+	processDecider: grimm.processDeciders.always(),
+	onSendError: console.err
+}
 const grimmUrl = 'http://grimm.example.com'
```
Comment (martinj) [inline README.md]: These aren't used any longer.

Comment (gunnarlium) [inline README.md]: Removed.

File: README.md [hunk 2]
```diff
@@ -13,11 +13,24 @@ Install with npm:
...
-const createGrimmService = require('@aptoma/grimm-pusher');
-
+const grimm = require('@aptoma/grimm-pusher');
+
+const options = {
+	host: 'http://grimm.example.com',
+	apiKey: 'secret',
+	// Use process decider to control when items are actually sent
+	processDecider: grimm.processDeciders.always(),
+	onSendError: console.err
```
Comment (martinj) [inline README.md]: Wouldn't  it better to call this `onError` instead so if we introduce more errors we won't need to add more options or rename.

Comment (gunnarlium) [inline README.md]: Maybe, I'm passing in the failed events as well. Might be good to have separate signatures depending on the error.

Comment (martinj) [inline README.md]: Distinguish errors with specific `code` or Error class is preferable and most common.

**Comment by gunnarlium** [comment]
Process deciders have been replaced with simpler batch and throttling support. I've also rewritten to use EventEmitter instead of configuring callbacks for various events.

The README should give a good overview of usage. As almost all the code has been changed, it's probably easier to just review the code as it is, rather than the diff from the old version.

**Comment by aptomaKetil** [comment]
LGTM

**Pull request merged**